cmd/pitr/collector/collector.go

```diff
@@ -205,54 +222,77 @@ func (c *Collector) manageBinlog(binlog string) (err error) {
 
 	tmpDir := os.TempDir() + "/"
 
-	err = os.Remove(tmpDir + binlog)
+	err = os.Remove(tmpDir + binlog.Name)
 	if err != nil && !os.IsNotExist(err) {
 		return errors.Wrap(err, "remove temp file")
 	}
 
-	err = syscall.Mkfifo(tmpDir+binlog, 0666)
+	err = syscall.Mkfifo(tmpDir+binlog.Name, 0666)
 	if err != nil {
 		return errors.Wrap(err, "make named pipe file error")
 	}
 
-	file, err := os.OpenFile(tmpDir+binlog, syscall.O_NONBLOCK, os.ModeNamedPipe)
+	err = os.Setenv("MYSQL_PWD", os.Getenv("PXC_PASS"))
+	if err != nil {
+		return errors.Wrap(err, "set mysql pwd env var")
+	}
+
+	cmd := exec.Command("mysqlbinlog", "-R", "--raw", "-h"+c.db.GetHost(), "-u"+c.pxcUser, "--result-file="+tmpDir, binlog.Name)
+
+	errOut, err := cmd.StderrPipe()
+	if err != nil {
+		return errors.Wrap(err, "get mysqlbinlog stderr pipe")
+	}
+
+	file, err := os.OpenFile(tmpDir+binlog.Name, os.O_RDWR, os.ModeNamedPipe)
 	if err != nil {
 		return errors.Wrap(err, "open named pipe file error")
 	}
+
 	defer func() {
 		errC := file.Close()
 		if errC != nil {
 			err = mergeErrors(err, errors.Wrapf(errC, "close tmp file for %s", binlog))
 			return
 		}
-		errR := os.Remove(tmpDir + binlog)
+		errR := os.Remove(tmpDir + binlog.Name)
 		if errR != nil {
 			err = mergeErrors(err, errors.Wrapf(errR, "remove tmp file for %s", binlog))
 			return
 		}
 	}()
-	err = os.Setenv("MYSQL_PWD", c.pxcPass)
-	if err != nil {
-		return errors.Wrap(err, "set mysql pwd env var")
-	}
 
-	cmd := exec.Command("mysqlbinlog", "-R", "--raw", "-h"+c.db.GetHost(), "-u"+c.pxcUser, "--result-file="+tmpDir, binlog)
-
-	errOut, err := cmd.StderrPipe()
-	if err != nil {
-		return errors.Wrap(err, "get mysqlbinlog stderr pipe")
+	pipeBuf := bytes.NewBuffer([]byte{})
+	pr := pipeReader{
+		f:   file,
+		buf: pipeBuf,
 	}
+	go pr.ReadToBuf()
 
 	err = cmd.Start()
 	if err != nil {
 		return errors.Wrap(err, "run mysqlbinlog command")
 	}
 
-	data := reader{file}
-	err = c.storage.PutObject(binlogName, &data)
+	for {
+		time.Sleep(10 * time.Millisecond)
+		if pr.notEmpty {
+			break
+		}
+		stdErr, err := ioutil.ReadAll(errOut)
+		if err != nil {
+			return errors.Wrap(err, "read mysqlbinlog error output")
+		}
+		if stdErr != nil && len(stdErr) != 0 {
```

Comment:
check for len is enough

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/723/commits/753da472c74c0b2c866b9648a610daa855c80cd8